features/step_definitions/user_steps.rb

```diff
@@ -151,22 +151,12 @@ def page_heading_text(page)
   click_on('Register')
 end
 
-Given('they login as {string}') do |username|
-  fill_in('username', with: username)
-  fill_in('password', with: 'bar')
-  click_on('SignIn')
-  click_on('I Agree')
-end
-
-Given('they login as the newly registered user') do
-  fill_in('username', with: @username)
-  fill_in('password', with: 'bar')
-  click_on('SignIn')
-  click_on('I Agree')
-end
+Given /they login as "(.*)"(?: with a random pid|)$/ do |userstring|
```

Comment:
Like Chris said, we need to detect whether `with a random pid` has been supplied and execute line 163: `page.execute_script('document.getElementById("randomPid").value = "true"')` only if it has.

Reply:
@CharlesIC  Changes made. Can you please re-review.